### ISSUE TITLE
Make ElementFinder(Driver, WebElementSource, By, int) accessible

### ISSUE THREAD

**Issue by Vuskk5** [selenide/selenide#1533]
## The problem
Before the changes in https://github.com/selenide/selenide/issues/1055, In my code I was calling `SelenidePageFactory#initElements` and was sending a `SelenideElement` as `SearchContext`.  
Now I no longer can do that since the changes forced me to use `WebElementSource`, so now I want to create an `ElementFinder` which will be sent to the initElements, I do that because I initialize my `ElementsContainer` with PageFactory.
`ElementFinder`'s constructor is package-private and thus I cannot initialize it.

If there is a better approach to this issue I would love to hear it.

Thanks in advance!

## Tell us about your environment
* **Selenide Version**: 5.22.1
* **Chrome\Firefox\IE Version**: 92.0.4515.131

## Code To Reproduce Issue [ Good To Have ]
```Java
@SneakyThrows
protected HtmlElement(By selfLocator, int index) {
    // Get the ElementFinder's package-private constructor and allow access
    Constructor<?> constructor = ElementFinder.class.getDeclaredConstructor(Driver.class, WebElementSource.class, By.class, Integer.TYPE);
    var isAccessible = constructor.canAccess(null);
    constructor.setAccessible(true);

    // Create a source for the 'self' context
    WebElementSource parent = (WebElementSource) constructor.newInstance(WebDriverRunner.driver(),
                                                                         null,
                                                                         new ByChained(selfLocator, Find.FindByBuilder.parseClassRootToLocator(this.getClass())),
                                                                         index);
    constructor.setAccessible(isAccessible);

    // Set the 'self' context
    Field selfField = ElementsContainer.class.getDeclaredField("self");
    isAccessible = selfField.canAccess(this);
    selfField.setAccessible(true);
    selfField.set(this, Selenide.$(parent.getWebElement()));
    selfField.setAccessible(isAccessible);

    Type[] types = this.getClass().getGenericInterfaces();

    // Init with factory
    new SelenidePageFactory().initElements(WebDriverRunner.driver(),
                                           parent,
                                           this,
                                           types);
}
```

**Comment by kfirdo** [comment]
This class is supposed to be my general element, which my wrappers (Element Containers) extend from. It provides the general functionality of elements, currently it has nothing in it, the code above was changed.

But I'll explain further, I attempt to create the element container explicitly in a static `$$` which I defined, the content of the c-tor in the example above is no longer there and was moved to said method.

The reason for this was an attempt to implement a `$$` which can also receive a class for casting, so I'll have a list of Containers.

This method was supposed to be used inside of containers and inside of pages, but it seems that while this works great at the highest context (Driver), when there is some hierarchy (Called within a container), it does not work.

After some experimenting it seems that it might be impossible since 'self' is getting its value after the object has been created, thus 'self' can't be referenced at field declaration, so every field must be lazy initialized and it is an over-complex solution for me, if even possible.

And yes, the easier solution is to use annotations.
